pkg/globalnet/controllers/cluster_egressip_controller.go

```diff
@@ -93,32 +90,52 @@ func NewClusterGlobalEgressIPController(config syncer.ResourceSyncerConfig, pool
 		return nil, err
 	}
 
+	iptIface, err := iptablesdriver.NewIPTableInterface()
+	if err != nil {
+		return nil, errors.WithMessage(err, "error creating the IPTablesInterface handler")
+	}
+
+	controller.iptIface = iptIface
+	controller.localSubnets = localSubnets
+
 	return controller, nil
 }
 
 func (c *clusterGlobalEgressIPController) process(from runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
 	clusterGlobalEgressIP := from.(*submarinerv1.ClusterGlobalEgressIP)
+	numberOfIPs := clusterGlobalEgressIP.Spec.NumberOfIPs
+	if numberOfIPs == nil {
+		one := 1
+		numberOfIPs = &one
+	}
+
+	klog.Infof("Processing %sd for %q, Spec.NumberOfIPs: %d, Status: %#v", op, clusterGlobalEgressIP.Name,
+		*numberOfIPs, clusterGlobalEgressIP.Status)
+
+	prevStatus := clusterGlobalEgressIP.Status
 
-	klog.Infof("Processing %sd %#v", op, clusterGlobalEgressIP)
+	if err := c.validate(numberOfIPs, clusterGlobalEgressIP); err != nil {
+		klog.Warningf("Error: %v", err)
+		return checkGlobalEgressIPStatusChanged(&prevStatus, &clusterGlobalEgressIP.Status, clusterGlobalEgressIP), false
+	}
```

Comment:
I don't think we want to do this validation for delete - we can't update the status anyway. Perhaps we shouldn't do anything for delete?

Also `allocateIPs` already validates the numberOfIPs.

 I think we should handle create and update the same, ie `onCreateOrUpdate`.  On a restart, if the object was changed while we were down, then it's the same as an update.